### PR TITLE
OCPBUGS#9122: Improved configuring your firewall install info

### PR DESCRIPTION
**Pull request by dfitzmau**
[OCPBUGS-9122](https://issues.redhat.com/browse/OCPBUGS-9122)

Version(s):
4.14 and 4.10

Link to docs preview:
* [Configuring your firewall for OCP](https://62848--docspreview.netlify.app/openshift-enterprise/latest/installing/install_config/configuring-firewall.html#configuring-firewall_configuring-firewall)
* [AWS firewall prerequisites](https://62848--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa_getting_started_iam/rosa-aws-prereqs.html#osd-aws-privatelink-firewall-prerequisites_prerequisites) 

QE review:
- [x] QE has approved this change.

Additional information:
* [Useful link for stakeholders](https://github.com/openshift/openshift-docs/pull/41850)

### CODE REVIEW

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 1]
```diff
@@ -57,9 +57,11 @@ This section provides the necessary details that enable you to control egress tr
 |443
 |Provides {op-system-first} images.
 
-|`registry.access.redhat.com`
+|`registry.access.redhat.com` ^[1]^
 |443
-|Provides access to the `odo` CLI tool that helps developers build on OpenShift and Kubernetes.
+|Hosts all the container images that are stored on the Red Hat Ecosytem Catalog. Additionally, the registry provides access to the `odo` CLI tool that helps developers build on OpenShift and Kubernetes.
+
+registry.redhat.io now?  https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/configuring_clusters/install-config-configuring-red-hat-registry
```
Comment (dfitzmau) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:64]: I see a note in the [old 3.11](https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/configuring_clusters/install-config-configuring-red-hat-registry#:~:text=The%20new%20registry%2C%20registry.redhat.io%2C%20requires%20authentication%20for%20access%20to%20images%20and%20hosted%20content%20on%20OpenShift%20Container%20Platform.) docs that states `registry.redhat.io` replaces `registry.access.redhat.com`?

Is this true? 

registry.redhat.io now? If so, do I need to swap terms?

**Comment by gpei** [comment]
@dfitzmau Hi, seem from the bug Description, it's about when we using `registry.access.redhat.com` registry, `access.redhat.com` also need to be allowed since signature store hosted on access.redhat.com.
But in this PR, I see we added
```
signature store that is hosted on `registry.access.redhat.com`.
```
so it confused me a bit, could you please clarify that?

**Comment by dfitzmau** [comment]
> access.redhat.com

Thanks, @gpei . This was my misunderstanding. I updated the content to state:

> In a firewall environment, ensure that the `access.redhat.com` resource is on the allowlist. This resource hosts a signature store that `registry.access.redhat.com` uses for verifying subscriptions before you can pull images from the registry.

Update based on reading https://access.redhat.com/RegistryAuthentication .

WDYT of this update.

**Comment by dfitzmau** [comment]
> > access.redhat.com
> 
> Thanks, @gpei . This was my misunderstanding. I updated the content to state:
> 
> > In a firewall environment, ensure that the `access.redhat.com` resource is on the allowlist. This resource hosts a signature store that `registry.access.redhat.com` uses for verifying subscriptions before you can pull images from the registry.
> 
> Update based on reading https://access.redhat.com/RegistryAuthentication . Article states "registry.access.redhat.com (no authentication needed)". I also see "some images that require a subscription are only available from registry.redhat.io.", so not too sure what the resource is needed by `registry.access.redhat.com`? Can you please clarify?

**Comment by gpei** [comment]
> > access.redhat.com
> 
> Thanks, @gpei . This was my misunderstanding. I updated the content to state:
> 
> > In a firewall environment, ensure that the `access.redhat.com` resource is on the allowlist. This resource hosts a signature store that `registry.access.redhat.com` uses for verifying subscriptions before you can pull images from the registry.
Thanks, the new change looks good to me based on the bug description 👍 

> 
> Update based on reading https://access.redhat.com/RegistryAuthentication . Article states "registry.access.redhat.com (no authentication needed)". I also see "some images that require a subscription are only available from registry.redhat.io.", so not too sure what the resource is needed by `registry.access.redhat.com`? Can you please clarify?

I'm also not sure about that, usually QE won't be involved into those image publishing things.

**Comment by dfitzmau** [comment]
#
/label peer-review-needed

**Comment by dfitzmau** [comment]
> > 
> 
> > > Update based on reading [access.redhat.com/RegistryAuthentication](https://access.redhat.com/RegistryAuthentication) . Article states "registry.access.redhat.com (no authentication needed)". I also see "some images that require a subscription are only available from registry.redhat.io.", so not too sure what the resource is needed by `registry.access.redhat.com`? Can you please clarify?
> > 
> > 
> > I'm also not sure about that, usually QE won't be involved into those image publishing things.
> 
> Did this get clarified?

I can get a second opinion. I'll ask Mike if an SME exists.

File: modules/configuring-firewall.adoc [hunk 2]
```diff
@@ -54,9 +54,14 @@ If your environment has a dedicated load balancer in front of your {product-titl
 |`sso.redhat.com`
 |443, 80
 |The `https://console.redhat.com/openshift` site uses authentication from `sso.redhat.com`
-
+[.small]
 |===
 +
+[.small]
+--
+1. In a firewall environment, ensure that the `access.redhat.com` resource is on the allowlist. This resource hosts a signature store that `registry.access.redhat.com` uses for verifying subscriptions before you can pull images from the registry.
```
Comment (patrickdillon) [inline modules/configuring-firewall.adoc:62]: Something like:
```suggestion
1. In a firewall environment, ensure that the `access.redhat.com` resource is on the allowlist. This resource hosts a signature store required to verify images when pulling from `registry.access.redhat.com` .
```

**Review by patrickdillon — COMMENTED** [review]
> > In a firewall environment, ensure that the `access.redhat.com` resource is on the allowlist. This resource hosts a signature store that `registry.access.redhat.com` uses for verifying subscriptions before you can pull images from the registry.
> 
> Update based on reading https://access.redhat.com/RegistryAuthentication . Article states "registry.access.redhat.com (no authentication needed)". I also see "some images that require a subscription are only available from registry.redhat.io.", so not too sure what resource is needed by `registry.access.redhat.com`? Can you please clarify?

@gpei is correct (good catch!): the signature store is used for verifying the container images--not the subscription.

File: modules/configuring-firewall.adoc (hunk 2 above)
Comment (dfitzmau) [reply modules/configuring-firewall.adoc]: Thanks. Updated to:

> In a firewall environment, ensure that the `access.redhat.com` resource is on the allowlist. This resource hosts a signature store that a container tool requires for verifying images when pulling them from `registry.access.redhat.com`.

**Review by jldohmann — APPROVED** [review]
LGTM

**Pull request merged**